src/Features/Core/Portable/MoveStaticMembers/MoveStaticMembersWithDialogCodeAction.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.ExtractInterface;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.MoveStaticMembers
+{
+    internal class MoveStaticMembersWithDialogCodeAction : CodeActionWithOptions
+    {
+        private readonly Document _document;
+        private readonly ISymbol? _selectedMember;
+        private readonly INamedTypeSymbol _selectedType;
+        private readonly IMoveStaticMembersOptionsService _service;
+
+        public TextSpan Span { get; }
+        public override string Title => "[WIP] Move Members to Type...";
+
+        public MoveStaticMembersWithDialogCodeAction(
+            Document document,
+            TextSpan span,
+            IMoveStaticMembersOptionsService service,
+            INamedTypeSymbol selectedType,
+            ISymbol? selectedMember = null)
+        {
+            _document = document;
+            _service = service;
+            _selectedType = selectedType;
+            _selectedMember = selectedMember;
+            Span = span;
+        }
+
+        public override object? GetOptions(CancellationToken cancellationToken)
+        {
+            var optionsService = _service ?? _document.Project.Solution.Workspace.Services.GetRequiredService<IMoveStaticMembersOptionsService>();
+            return optionsService.GetMoveMembersToTypeOptions(_document, _selectedType, _selectedMember);
```

Comment:
This never sets `_ service` so you don't get any benefit of caching the lookup.

```suggestion
            _service ??= _document.Project.Solution.Workspace.Services.GetRequiredService<IMoveStaticMembersOptionsService>();
            return _service.GetMoveMembersToTypeOptions(_document, _selectedType, _selectedMember);
```